src/cpp/dds/core/policy/QosPolicies.cpp

```diff
@@ -593,14 +612,13 @@ bool TransportPriorityQosPolicy::readFromCDRMessage(
 }
 
 bool DataRepresentationQosPolicy::addToCDRMessage(
-        CDRMessage_t* msg)
+        CDRMessage_t* msg) const
 {
     bool valid = CDRMessage::addUInt16(msg, this->Pid);
-    this->length = 4;
-    this->length += static_cast<uint16_t>(m_value.size() * sizeof(uint16_t));
-    valid &= CDRMessage::addUInt16(msg, this->length);
+    uint16_t len = 4 + static_cast<uint16_t>(m_value.size() * sizeof(uint16_t));
```

Comment:
Should account for alignment (see line 625 below). We should also add a `cdr_serialized_size` override for this class.